### ISSUE TITLE
Same image reference in multiple entries processed only once

### ISSUE THREAD

**Issue by benjaminpreiss** [webpack-contrib/html-loader#290]
- Operating System: Windows 10 Professional
- Node Version: 14.3.0
- NPM Version: 6.14.5
- webpack Version: 4.43.0
- html-loader Version: 1.1.0

### Expected Behavior

A image that is referenced multiple times in different html files/entries should be passed to its corresponding loader multiple times.

In my webpack config I have 8 entries that are different html files all referencing the same svg in the header.

Also I configured a file-loader to do the following:

1. load the referenced images
2. supply a custom public path for each image depending  on the `info.issuer`
3. log the issuer to the console.

Therefore I expect to see 8 logs in the console stating 8 issuers (the 8 html files)

### Actual Behavior

I can only see one log in the console stating 1 issuer. This issuer seems to be randomly chosen out of the 8 entry points.

Also the public path is the same through all generated html files.

As you can see in the provided lemon.html file the svg url in the `header` is wrong. It should be `../svg/f20858e92f4c08e894ddfd7d59eca9a7.svg`

### Code

**Generated lemon.html file:**
```html
<!doctype html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>CHURCHILL</title>
    <link href="../css/lemon.css" rel="stylesheet">
</head>

<body data-barba="wrapper" class="lemon.html">
    <main data-barba="container" data-barba-namespace="lemon" class="m-lemon">
        <div class="m-media-canvas">
            <div class="m-image-container">
                <div class="m-image"></div>
            </div><canvas class="m-canvas" resize="true"></canvas>
        </div>
        <div class="m-text-container">
            <div class="m-text-table">
                <h1>Lemon</h1>
                <div class="m-paragraphs">
                    <p><span>Our lemons provide CHURCHILL with its</span> <span>refreshing flavor and boost your immune
                            defense.</span> <span>Additionally, they add up to 30% of the daily</span> recommended
                        Vitamin C Dosis.</p>
                    <p><span>Through our pasteurization process, we</span> <span>eliminate the fear of bacteria while
                            specifically</span> <span>paying attention to ensure that our all-natural</span> vitamins
                        remain intact.</p>
                </div>
            </div>
        </div>
    </main>
    <header class="m-header"><button class="m-logo-header-link"><img src="./svg/f20858e92f4c08e894ddfd7d59eca9a7.svg"
                alt="Churchill logo simple grey"></button>
        <div class="m-questionaire-link"></div><button class="m-navigation-toggle">
            <div class="m-toggle-bar"></div>
            <div class="m-toggle-bar"></div>
            <div class="m-toggle-bar"></div>
        </button>
    </header>
    <footer class="m-footer">
        <div class="m-privacy-toggle">
            <h5>Privacy</h5>
        </div>
        <div class="m-social-media-link">
            <div class="m-insta-link"></div>
            <div class="m-facebook-link"></div>
        </div>
        <div class="m-imprint-toggle">
            <h5>Imprint</h5>
        </div>
    </footer>
    <script src="../js/lemon.bundle.js"></script>
</body>

</html>
```

### How Do We Reproduce?

Here is my repo with the webpack config files:

[https://github.com/benjaminpreiss/churchill_website/tree/v2.0.0](https://github.com/benjaminpreiss/churchill_website/tree/v2.0.0)

**Comment by evilebottnawi** [comment]
> A image that is referenced multiple times in different html files/entries should be passed to its corresponding loader multiple times.

No, It should not behave this way, it's not only getting bundle to be huge, but reduce perf. I will say more, the webpack@5 will throw an error on this. `scss` entrypoint will never officially supported https://github.com/benjaminpreiss/churchill_website/blob/v2.0.0/webpack.config.js#L15, yes it is works but with limitation, custom entry points will be implemented in webpack@5 too.
Why do you need it?

**Comment by benjaminpreiss** [comment]
@evilebottnawi Thanks for the fast answer!

I need it because my index.html file and the lemon.html file are in different folders in `dist`... But in src they are the same. So I need to adjust the public path in file-loader depending on the entrypoint.

If the entrypoint is index.html I need the public path to be `./svg/`
If the entrypoint is lemon.html or any other html file I need the public path to be `../svg/`.

I do not want webpack to output the files twice. I just want to adjust the public path depending on entrypoint...

**Comment by evilebottnawi** [comment]
Sorry, it is impossible, as I said early in webpack@5 you will get a compilation error on multiple same assets and it is expected. I think you need to review the structure of the application. Unfortunately, I can’t give the correct answer, as this will require a detailed study of the project. 

Ideas:
- Maybe avoid using `HtmlWebpackPlugin` and using `src/html/lemon.html`/`src/html/mint.html`/etc and use https://github.com/peerigon/extract-loader to extract it.
- Using multi-compile mode for each page
- Just don't use relative paths relative to the `index.html`/`lemon.html` file, because it is a bad idea, `publicPath: () => {}` is not for this, if you need relative paths relative html documents, you need to use a plugin and analyses all assets and rewrite paths. It is not easy as you think. That is why we never recommend doing this.

Sorry, it is limitation and won't fix for webpack@4, we want to implement built-in html support on entry points for webpack@5, so your problem can be solved in right way. Right now I recommend not using relative paths relative to `html` file.
